larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/ShowerNumElectronsEnergy_tool.cc

```diff
@@ -111,108 +91,61 @@ namespace ShowerRecoTools {
         clusHandle, Event, fPFParticleLabel);
     // art::FindManyP<recob::Hit> fmhc(clusHandle, Event, fPFParticleLabel);
 
-    std::vector<std::vector<art::Ptr<recob::Hit> > > trackHits;
-    trackHits.resize(numPlanes);
-
+    std::map<unsigned int, std::vector<art::Ptr<recob::Hit> > > planeHits;
 
     //Loop over the clusters in the plane and get the hits
     for(auto const& cluster: clusters){
 
       //Get the hits
       std::vector<art::Ptr<recob::Hit> > hits = fmhc.at(cluster.key());
-      if(hits.empty()){
-        if (fVerbose)
-          mf::LogWarning("ShowerNumElectronsEnergy") << "No hit for the cluster. This suggest the find many is wrong."<< std::endl;
-        continue;
-      }
 
-      //Get the view. geo::View_t view = cluster->View();
-      geo::View_t view = cluster->View();
-      //std::cout << "view = " << view << " hits = " << hits.size() << std::endl;
-
-      view_hits[view].insert(view_hits[view].end(),hits.begin(),hits.end());
+      //Get the plane.
+      unsigned int plane = cluster->Plane().Plane;
 
+      planeHits[plane].insert(planeHits[plane].end(),hits.begin(),hits.end());
     }
 
-    std::map<unsigned int, double > view_energies;
-    std::vector<art::Ptr<recob::Hit>> hits;
+    // Calculate the energy fro each plane && best plane
+    int bestPlane                 = -999;
```

Comment:
`Plane_t` typedef.
Some thought might be needed here as I believe the typedef points to a `unsigned int` so -999 wouldn't be a useful default value.